### PR TITLE
Implement datetime and Junit tests

### PR DESCRIPTION
**Pull request by shashd**
co-author: gustav

### CODE REVIEW

**Review by avinkadir — CHANGES_REQUESTED** [review]
Great job! The code looks great overall but we just have a few suggestions to further improve it. First, we saw that code coverage is a bit low but we understand that it would be difficult to get a high coverage. Perhaps you could write a few tests for today() and __str__ to improve it?

File: junit_tests/DateTimeTest.java [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+    @Test
+    public void testConstructor() {
+        // use args and empty kwargs to create instance
+        Object[] args = {Int.getInt(2021), Int.getInt(9), Int.getInt(18),
+            Int.getInt(14), Int.getInt(0), Int.getInt(0), Int.getInt(0 / 1000)};
+        DateTime dateTime = new DateTime(args, Collections.emptyMap());
+        HashMap<String, Object> kwargs = new HashMap<>();
+        testVariables(dateTime, args, kwargs);
+
+        // use empty args and kwargs to create instance
```
Comment (yuch3628) [inline junit_tests/DateTimeTest.java:67]: Great comments ! We think they make it easier to understand what you are testing.

File: python/common/org/python/stdlib/datetime/DateTime.java [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+                return org.python.types.Bool.getBool(false);
+            }
+            if (((org.python.types.Bool)this.__year__().__le__(((DateTime) other).__year__())).value &&
+                ((org.python.types.Bool)this.__month__().__le__(((DateTime) other).__month__())).value &&
+                ((org.python.types.Bool)this.__day__().__le__(((DateTime) other).__day__())).value &&
+                ((org.python.types.Bool)this.__hour__().__le__(((DateTime) other).__hour__())).value &&
+                ((org.python.types.Bool)this.__minute__().__le__(((DateTime) other).__minute__())).value &&
+                ((org.python.types.Bool)this.__second__().__le__(((DateTime) other).__second__())).value &&
+                ((org.python.types.Bool)this.__microsecond__().__le__(((DateTime) other).__microsecond__())).value){
+                return org.python.types.Bool.getBool(true);
```
Comment (yuch3628) [inline python/common/org/python/stdlib/datetime/DateTime.java]: This code block is not necessary, the method works without it. We think removing it improves the readability of the code.

File: python/common/org/python/stdlib/datetime/DateTime.java [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+            return org.python.types.Bool.getBool(false);
+        }
+        return org.python.types.NotImplementedType.NOT_IMPLEMENTED;
+    }
+
+    @org.python.Method(
+        __doc__ = "Return self > dateTime.",
+        args = {"other"}
+    )
+    public org.python.Object __gt__(org.python.Object other){
```
Comment (yuch3628) [inline python/common/org/python/stdlib/datetime/DateTime.java:340]: Good job on this code!

**Review by yuch3628 — CHANGES_REQUESTED** [review]

**Comment by gustavborestam** [comment]
We've added more tests to improve coverage and also simplified Le and Ge functions to make them more readable.

Coverage from 67% -> 89%

**Review by yuch3628 — APPROVED** [review]

**Review by avinkadir — APPROVED** [review]

**Pull request merged**